### ISSUE TITLE
PostGIS Error on init

### ISSUE THREAD

**Issue by deniszgonjanin** [datacats/datacats#195]
Doing a `datacats init` on an existing project with `ckanext-spatial` installed causes PostGIS error

```
Initializing database
----------------------------------------
datacats: problem was encountered:
Something that should not have happened happened when attempting to run this command:
     datacats init

It is seems to be a bug.
Please report this issue to us by creating an issue ticket at

    https://github.com/datacats/datacats/issues

so that we would be able to look into that and fix the issue.


Please quote this traceback when reporting this error:
Traceback (most recent call last):
  File "/Users/dz/source/datacats/datacats/cli/main.py", line 89, in main
    return command_fn(opts)
  File "/Users/dz/source/datacats/datacats/cli/create.py", line 189, in init
    log_syslog=log_syslog, do_install=not no_install)
  File "/Users/dz/source/datacats/datacats/cli/create.py", line 201, in finish_init
    environment.ckan_db_init()
  File "/Users/dz/source/datacats/datacats/environment.py", line 634, in ckan_db_init
    clean_up=True,
  File "/Users/dz/source/datacats/datacats/environment.py", line 1031, in run_command
    commit=True, stream_output=stream_output)
  File "/Users/dz/source/datacats/datacats/docker.py", line 200, in web_command
    raise WebCommandError(command, c['Id'][:12], logs)
WebCommandError:     Command: / u s r / l i b / c k a n / b i n / p a s t e r   - - p l u g i n = c k a n   d b   i n i t   - c   / p r o j e c t / d e v e l o p m e n t . i n i
    Docker Error Log:
    2015-06-15 18:28:57,796 DEBUG [ckanext.issues.model] Issue tables defined in memory
2015-06-15 18:28:57,798 DEBUG [ckanext.issues.model] Issue Extension tables already exist
2015-06-15 18:28:57,798 DEBUG [ckanext.showcase.model] ShowcasePackageAssociation table defined in memory
2015-06-15 18:28:57,801 DEBUG [ckanext.showcase.model] ShowcasePackageAssociation table creation deferred
2015-06-15 18:28:57,802 DEBUG [ckanext.showcase.model] ShowcaseAdmin table defined in memory
2015-06-15 18:28:57,804 DEBUG [ckanext.showcase.model] ShowcaseAdmin table creation deferred
Traceback (most recent call last):
  File "/usr/lib/ckan/bin/paster", line 9, in <module>
    load_entry_point('PasteScript==1.7.5', 'console_scripts', 'paster')()
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/paste/script/command.py", line 104, in run
    invoke(command, command_name, options, args[1:])
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/paste/script/command.py", line 143, in invoke
    exit_code = runner.run(args)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/paste/script/command.py", line 238, in run
    result = self.command()
  File "/project/ckan/ckan/lib/cli.py", line 208, in command
    self._load_config(cmd!='upgrade')
  File "/project/ckan/ckan/lib/cli.py", line 148, in _load_config
    load_environment(conf.global_conf, conf.local_conf)
  File "/project/ckan/ckan/config/environment.py", line 232, in load_environment
    p.load_all(config)
  File "/project/ckan/ckan/plugins/core.py", line 134, in load_all
    load(*plugins)
  File "/project/ckan/ckan/plugins/core.py", line 166, in load
    plugins_update()
  File "/project/ckan/ckan/plugins/core.py", line 116, in plugins_update
    environment.update_config()
  File "/project/ckan/ckan/config/environment.py", line 361, in update_config
    plugin.configure(config)
  File "/project/ckanext-spatial/ckanext/spatial/plugin.py", line 78, in configure
    setup_model()
  File "/project/ckanext-spatial/ckanext/spatial/model/package_extent.py", line 22, in setup
    define_spatial_tables(srid)
  File "/project/ckanext-spatial/ckanext/spatial/model/package_extent.py", line 69, in define_spatial_tables
    package_extent_table = setup_spatial_table(PackageExtent, db_srid)
  File "/project/ckanext-spatial/ckanext/spatial/geoalchemy_common.py", line 68, in setup_spatial_table
    management = (postgis_version()[:1] == '1')
  File "/project/ckanext-spatial/ckanext/spatial/geoalchemy_common.py", line 40, in postgis_version
    result = Session.execute('SELECT postgis_lib_version()')
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 149, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 981, in execute
    clause, params or {})
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 727, in execute
    return meth(self, multiparams, params)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 322, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 824, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 954, in _execute_context
    context)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1116, in _handle_dbapi_exception
    exc_info
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 189, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 947, in _execute_context
    context)
  File "/usr/lib/ckan/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 435, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (ProgrammingError) function postgis_lib_version() does not exist
LINE 1: SELECT postgis_lib_version()
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
 'SELECT postgis_lib_version()' {}

----------------------------------------
```

**Comment by deniszgonjanin** [comment]
Actually, same happens with new environments. `datacats create` works fine, but when ckanext-spatial is installed and added to ckan.plugins, `datacats reload` reports the same error.

**Comment by deniszgonjanin** [comment]
@JackMc perhaps the commands in https://github.com/datacats/datacats/blob/master/datacats/scripts/install_postgis.sh can simply be added to https://github.com/datacats/datacats/blob/master/docker/postgres/ckan.sh? I know @wardi has some qualms about this, but I would be very happy with something that just works at this very moment

**Comment by wardi** [comment]
this part is fishy:
```
WebCommandError:     Command: / u s r / l i b / c k a n / b i n / p a s t e r   - - p l u g i n = c k a n   d b   i n i t   - c   / p r o j e c t / d e v e l o p m e n t . i n i
```
did someone call list() on command before passing it in?

**Comment by JackMc** [comment]
Ran into that before, there's a .join(" ") in webcommanderror

**Comment by wardi** [comment]
@deniszgonjanin feel free to modify the images on peach if that makes your life easier, that server is all yours for now.

**Comment by dborzov** [comment]
yeah, it is `" ".join(WebCommandError.message)`, should be fixed

**Comment by dborzov** [comment]
Error formatting issue is now fixed in [#194](https://github.com/datacats/datacats/pull/194)

**Comment by deniszgonjanin** [comment]
Fixed by #198
